File: frontend/src/microservices/advertising/objects/UploadAdvertisementImageRequest.ts
// 本文件定义 advertising 模块的 `UploadAdvertisementImageRequest`，用于图片上传入参并提供 JSON 编解码。

export type UploadAdvertisementImageRequest = {
  originalFileName: string
  mimeType: string
  fileContentBase64: string
}

export const uploadAdvertisementImageRequestFromJson = (json: string): UploadAdvertisementImageRequest =>
  JSON.parse(json) as UploadAdvertisementImageRequest

export const uploadAdvertisementImageRequestToJson = (value: UploadAdvertisementImageRequest): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/advertising/objects/UploadAdvertisementImageRequest.scala
// 本文件定义 advertising 模块的 UploadAdvertisementImageRequest，作为后端与前端同名的数据模型，并提供 JSON codec。
package com.typesafe.travel.advertising.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class UploadAdvertisementImageRequest(
    originalFileName: String,
    mimeType: String,
    fileContentBase64: String
)

object UploadAdvertisementImageRequest:
  given sourceEncoder: Encoder[UploadAdvertisementImageRequest] = deriveEncoder
  given sourceDecoder: Decoder[UploadAdvertisementImageRequest] = deriveDecoder

